os/arch/arm/src/stm32l4/stm32l4_ospi.c

```diff
@@ -1921,7 +1921,7 @@ FAR struct mtd_dev_s *up_flashinitialize(void)
     priv->read = this_read;
     priv->ioctl = this_ioctl;
 #ifdef CONFIG_MTD_REGISTRATION
-    priv->mtd.name = "stm32l4_flash";
+    priv->name = "stm32l4_flash";
```

Comment:
This is not `Fix __sint_heap_start undeclared issue`. If this is necessary to success a build, let's title like `.....: fix build breaks on stm board`.